### DISCUSSION TITLE
v3.13.0

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#10129] [Releases & Announcements]
## Changelog

- Custom Roles (alpha) [EE]
- Admin page redesigns for macros, automation, inbox management, etc
- Support for copy-pasting images in help center articles 
- Update GPT model for open AI integration to 4o-mini
- Numerous bug fixes and enhancements

<hr /><em>This discussion was created from the release <a href='https://github.com/chatwoot/chatwoot/releases/tag/v3.13.0'>v3.13.0</a>.</em>

**Comment by fhonline** [comment]
The Custom Roles (alpha) [EE] being Paid feature was a shot in the foot... it will lose many users because of this, especially because, as Chatwoot's biggest focus is integration with WhatsApp, it became practically like WhatsApp Web, with "slight" modifications... if this resource were free, the community would be stronger, so they could sell premium resources in the future. Make features that make a difference premium and that are an extra service, not something basic that you had issues over 3 years old on this subject.

**Reply by jcmatoskx** [reply]
Yes, I feel like chatwoot should reconsider this. This is probably one of the most requested features here and it is a great feature that would set them apart. 
We would like to use ChatWoot because we are a small growing business and the lower costs of self hosting chatwoot justify it and allow usto have a multi channel ticket management platform. However in order to grow, role management is a feature that we need.
The thing is, as of now considering Chatwoot pricing per user and the existing main competitors/alternatives that are not open/self hosted (which is a plus for us as it would be one less resource to manage) and are also more mature, it feels like it makes more sense to, if we have to start paying because of this feature, adopt those instead. 
As you said either fully make this resource free in order to grow Chatwoot userbase even more or at least have Chatwoot allow Custom Roles for up to 5 or up to 10 users, but more than that make it a paid feature.

**Comment by fernandescontabilidade** [comment]
After updating, my plan changed from community to premium and I can't return to community. I believe it was a bug,

**Reply by sojan-official** [reply]
@fernandescontabilidade  could you ping us on the Chat so that the team can take a look?

**Reply by fernandescontabilidade** [reply]
Resolved

**Reply by aman400** [reply]
Facing a similar issue

**Reply by bernapastorini** [reply]
@aman400 Go to Super Admin Console and then Settings > Current Plan > Manage > Cancel

**Comment by meguisouza** [comment]
I kindly ask you to reconsider the decision to charge for the "Custom Roles (alpha) [EE]" feature and make it free. Making it accessible to everyone would strengthen the community and create more opportunities to offer premium features in the future. Basic functionalities like this should be free, especially considering the challenges faced on this topic over the past 3 years, as our friend @fhonline mentioned above.

**Comment by leofox1405** [comment]
Chatwoot is a very well-built platform, however, as our friends mentioned above, it seems like they don't pay much attention to the potential in the Brazilian market... For our reality, where we predominantly use Chatwoot with WhatsApp, not being able to have privacy control over conversations is indeed like using WhatsApp Web with just a few extra features... We ask that you reconsider the situation, as there is a huge market where the community will surely contribute more to the popularization of the Chatwoot brand.

This feature has been a basic request for years, and it would completely change the way we use the application. Keep in mind that your currency compared to ours is five times more expensive; how could we maintain a user that would cost almost half of a national minimum wage? Please pay closer attention to the potential in the Brazilian market...

**Reply by eronGreco** [reply]
Indeed.

**Comment by stactus** [comment]
I understand the need for a sustainable business model, but making **'Custom Roles' free** could foster stronger community support. By offering **essential features** without charge, you can build trust and encourage users to stay engaged. 

Over time, this approach would likely lead to a **higher willingness to migrate** to paid plans for premium features as their needs grow, resulting in better **long-term customer retention** and growth.

**Comment by edisonmartinsmkt** [comment]
I believe making **'Custom Roles' free** would be a great move to strengthen community loyalty. 

Offering **core features** without cost creates a sense of goodwill, leading to increased trust and satisfaction. As users grow and require more advanced functionalities, they will be more likely to consider **upgrading to a paid plan**. 

This strategy not only encourages adoption but also ensures **long-term growth** and customer retention, benefiting both the community and the platform in the long run.

**Comment by heslanx** [comment]
The decision to make the "Custom Roles (alpha) [EE]" a paid feature in Chatwoot v3.13.0 feels like a misstep. This is a fundamental feature that many of us have been waiting for, and locking it behind a paywall risks alienating users, especially small businesses. For those of us who rely on Chatwoot’s affordability and open-source nature, this change is disappointing.

By making essential features like role management free, Chatwoot could strengthen its community and encourage wider adoption. As businesses grow, they’ll naturally gravitate toward premium features, creating a more sustainable path for both users and the platform. Charging for such a basic feature feels counterproductive and could push users to seek alternatives.

I hope Chatwoot rethinks this. This would foster greater trust and loyalty, while still allowing for monetization of advanced, premium features in the future.

**Comment by fernandescontabilidade** [comment]
Here are 10 reasons why Chatwoot shouldn’t charge for custom privacy rules in conversations, considering it’s an essential feature:

1. Essential for legal compliance: Privacy rules are essential for companies to comply with data protection laws, such as GDPR and LGPD. Charging for them can put companies at legal risk.

2. Need for security: Privacy is not just an extra feature; it’s a basic requirement to ensure the security of customer data and prevent leaks or unauthorized access.

3. Improved user experience: Custom privacy rules ensure that customer conversations are handled appropriately, increasing trust in the service and the company. Charging for them can undermine this trust.

4. Avoiding barriers to entry: Small businesses and startups, which rely on affordable solutions like Chatwoot, can be hurt by extra charges, making it difficult to use the system in compliance with good privacy practices.

5. Preserving the open-source mission: Chatwoot is known as an open-source solution, and charging for essential features could go against that philosophy, alienating the community of developers who support the project.

6. Competitiveness with other platforms: Many of Chatwoot’s competitors offer personalized privacy at no additional cost. Charging for this could make the platform less competitive in the market.

7. Protecting customer data: Businesses rely on Chatwoot to protect their customers’ conversations and data. A feature like this should be available to everyone, regardless of the plan they sign up for.

8. Impact on Chatwoot’s adoption: Charging for essential features could discourage new businesses from adopting Chatwoot, hurting the platform’s growth and user base.

9. Aligning with market trends: Data privacy is a growing global priority. Offering this feature for free would help Chatwoot align with market trends and customer needs.

10. Overall customer support performance: Without custom privacy rules, companies may not be able to configure their workflows properly, hampering customer service efficiency, which can negatively impact the reputation of both the company and the platform.

**Comment by fernandescontabilidade** [comment]
Here are 10 reasons why the price charged by Chatwoot (19 dollars per user) should be revised to reflect the reality in Brazil:

1. Significant exchange rate difference: The value of $ 19 dollars, when converted to reais, exceeds R$ 90 reais per user, which is a high amount considering the reality of many Brazilian companies, especially small and medium-sized ones.

2. Minimum wage in Brazil: The current minimum wage in Brazil is R$ 1,412.00, which means that the price charged by Chatwoot represents a significant part of the budget of many companies, especially in sectors that operate with reduced margins.

3. High cost of living: Brazil has high costs in several areas (taxes, rent, transportation, etc.), making it difficult for smaller companies to afford additional tools, especially when there are competitors offering more affordable or free solutions.

4. Unequal economy: The Brazilian economy has significant regional disparities, with many areas of the country where purchasing power is significantly lower than in large cities, making the $19 price unviable for many companies.

5. Incentive to the local market: By adjusting the price to the Brazilian market, Chatwoot could encourage the use of the platform in one of the largest markets in Latin America, promoting the growth of its user base and consolidating its presence.

6. Competitiveness with national solutions: In Brazil, there are several local alternatives for customer support tools with more affordable prices. Keeping the price high could cause companies to opt for national solutions, harming Chatwoot's penetration in the market.

7. Difficulty of access for startups and small businesses: Many Brazilian startups and small businesses work with tight budgets and low profit margins, making it difficult to justify investing in a tool with a high cost per user.

8. Mass service companies: Brazilian businesses that need a large number of agents for customer service (such as call centers or e-commerces) may find the price prohibitive, considering the number of users who need access to the tool.

9. Economic crisis and inflation: The Brazilian economy often goes through cycles of recession and high inflation, which reduces the purchasing power of companies and makes it even more difficult to pay for tools in dollars.

10. Digital and social inclusion: More affordable prices would encourage smaller companies and social organizations to adopt Chatwoot, promoting digital inclusion and improving the quality of customer service in sectors that serve vulnerable populations.

Revising prices based on local purchasing power could strengthen the adoption of Chatwoot in Brazil, creating a broader and more loyal user base.

**Comment by mirmoh3n** [comment]
I'm also waiting for the Live Visitor feature like on tawkto,
I'm used to using tawkto and monitoring visitor traffic, when using chatwoot I feel strange because I'm not used to it and have to daydream without seeing the traffic that is happening on the website, in tawkto I can monitor the visitor currently accessing the page.
hopefully this feature will be available soon

**Comment by sanchez777** [comment]
I also would like to add that if permissions can't be made a free resource, then at least there should be basic permissions for conversations, such as choosing between allowing agent users to view all conversations, to view only those from the same team or to view only theirs.

**Comment by pranavrajs** [comment]
Hey everyone, 

I'd like to share our understanding of the feature and the thought process behind building roles and permissions addressing the concerns raised. 

**A bit on the feature**: In all fairness, the feature has been marked as enterprise from the day the issue was created and we have been looking at this feature as an option for larger companies, specifically for compliance, regulatory requirements. 

Everything we designed for small businesses or the features that are used by agents directly has been generally kept under the open-source version. For small businesses, granular access is not really required (at least from what we have understood talking to people). I would be happy to be proven wrong. Happy to learn the detailed use cases on how you would using the feature in small business and  the strong need for such capabilities. I agree that there is a strong community engagement on this issue, however it doesn’t change the understanding above. 

I'll explain my understanding of the requests for the feature to be in open-source (learned by reading the comments and talking to people with specific use cases). Some companies don't want agents to see the conversations other agents are handling. This, to my understanding, is from companies who incentivize people based on the number of conversations they close/attend. This leads people to grab tickets from other people. The way we see this scenario is as an anomaly, not a regular scenario, and is specific to companies in certain regions. This is more of a trust problem. I'm not downplaying the need, and I understand the case, but we can't build system standards based on low trust. You would see this everywhere in our product decisions. If it was low trust, we would have kept agent permissions to see the conversations assigned to them only. We want the software to promote collaboration and increase productivity for teams. Doing the reverse is not a good place for us. 

Now, there may be a specific set of permissions everyone wants badly (including the one described above). We can rethink making these permissions available on open-source somehow (we still need to investigate the engineering effort). Please pick the permissions you need badly and share the use cases you're going to solve; we will reconsider those. A blanket statement that you need it might not be helpful—we would appreciate the reasoning behind it.

Again the feature is for compliance, not for anything else, hence very much suited for enterprise edition.

**A bit on pricing**: On our Cloud, we would offer this feature from the Business plan onwards. For self-hosted solutions, it's available with the Premium Support plan and above. We understand that dollar-based pricing might be steep for some countries, so we're happy to offer significant discounts. Please reach out to our support team, and we'll work something out until we have a region-based pricing in place.

Let me know your thoughts.

**Reply by jcmatoskx** [reply]
Thanks for expanding on this and opening an opportunity to hear feedback. 

**I think what most people would like to setup is that Agents would NOT have access to an entire inbox and instead see ONLY conversations assigned to them from that inbox.**


**Our use case:**
In our case, as a small business (right now about 6 people, where 2 do support) we have one email (e.g. info@ourbusiness.com) 
Forever we have only had this one email for everything. Customer support, purchases, inquiries, creating our company accounts, sales, talking with suppliers etc.
This is simple, one email is easy to share with customers, easy to add to contacts when we create new accounts or social media, easy to manage, simple and easy to take care of.  

This was all good until our customer support kept increasing and increasing and now we have the need to expand our team in order to keep up with customer support.

As we introduce new team members to do customer support, it doesn't make sense to immediately give full access to our inbox. Many important emails with a certain degree of privacy fall on this inbox. Password resets for important accounts/accesses could be requested from this inbox.
These are new people to the company, they are learning and having access to the entire company inbox like this would be overwhelming and worse could be damaging to the company.
This way we could have much bigger peace of mind and scale our customer support faster, by assigning only specific customer support conversations manually (and then later automatically through automations).
Also we (as admins/managers) know the level of knowledge each customer support person has and the types of problems each can solve. 

Access permissions would solve this problem swiftly, quickly and effectively and would allow us to keep scaling our current system.
It would also allow us to scale our customer support to other platforms (facebook, instagram etc) while keeping everything organized in one place. 

There are other solutions we have thought of but everything else is less effective and has more downsides. 

- We could create a new internal email and change all of our important business accounts (bank accounts, services etc) to that email, then also request all our suppliers and important contacts to stop emailing our general email and start using our new internal email. This would be a lot of work and not efficient, as we have hundreds of contacts they would certainly keep emailing us on the wrong email.

- We could create more emails such as "support@ourbusiness.com", "sales@ourbusiness.com" etc... but that doesn't make sense because it only adds more complexity to day to day operations, it also relies and puts responsibility on the customers/users to choose the right email (they never do, or they end up sending the same email to all of them), and then requires us to redirect emails internally. 
This also makes it more difficult to have flexibility in the future, what if we want to retire a team like the Sales team? We would have to start redirecting the "sales@ourbusiness.com" email? What if we would want to create a new team? What if we want to do support in other platforms that are not email as well? It is really not a good solution.

Being able to use permissions and roles would be far more flexible, scalable, and powerful.
We started using Chatwoot because of the potential of and accessible way of centralizing all platform customer support in one place (social media and email) and simplifying our life, given the open source/self-hosting possibility. 
However self-hosting naturally also comes with downsides to us for needing to manage it ourselves and possibly having to deal with issues/errors/bugs, and with the need of paying per user for this feature it at this point it weighting all the pros/cons it pushes us to pay for other existing SAAS solutions that already have team permission features in place.
However we understand you also need to make money somewhere thus my suggestion above of possibly allowing permissions up to X amount of users ( such as 5) and more than that start to charge as it is already a bigger operation.  
Or allow at least the possibility to setup the following permissions as that is what I think most starting businesses like us need.
**_User can see only conversations assigned to them but not the entire inbox_**

**Reply by fernandescontabilidade** [reply]
Hello everyone,

I would like to share my perspective on creating roles and permissions in the context of small businesses and how this impacts the open source ecosystem.

First, I understand that the feature was originally intended for large enterprises, with a focus on compliance and regulatory requirements. However, the assumption that small businesses do not need granular permission control seems generalized to me and may not reflect the reality of many of these organizations. Small businesses, especially those that deal with sensitive data, can benefit immensely from more granular control over who has access to certain information and functionality. The need to ensure data security and privacy is not exclusive to large corporations; it is also vital for small and medium-sized businesses that have to comply with legislation such as GDPR, LGPD, among others.

In addition, the "low trust" issue that was raised deserves more careful attention. I do not see permission control as an issue of lack of trust, but rather as a way to establish clear boundaries between roles, ensuring that processes are followed in an appropriate and organized manner. This type of control promotes greater clarity in responsibilities and avoids misunderstandings or disputes between collaborators, something that can be useful in any work environment, regardless of the size of the company.

Regarding the argument that "we want software to promote collaboration", I completely agree, but I see that collaboration and control are not mutually exclusive. Collaborative systems can work perfectly with adequate permission control, which ensures that everyone knows exactly what their role is within a process. By providing more control options for small companies, we would be helping to promote more efficient and secure collaboration, without losing focus on the values ​​that the software seeks to defend.

Regarding the availability of the feature in specific editions, I understand the need to support development through paid plans, but I believe that limiting features such as permission control could directly impact the adoption of the platform by small companies and open source communities, which can be a significant and highly engaged user base. A more flexible approach, with a basic set of permissions in open source, could benefit all parties and further promote the adoption of the software.

Finally, the proposal to provide regional discounts is a great initiative, but I would like to suggest that more flexible policies on critical features such as permissions be reconsidered so that there is no barrier between small and large businesses when it comes to accessing tools that can be crucial to operational efficiency.

I look forward to continuing this discussion and sharing more details about specific use cases in the small businesses we work with.

Thank you!

**Reply by MarcioLacerda** [reply]
I'm not very experienced with Chatwoot, but maybe I can help with a real example that happened to me.

A small law firm made up of just 2 lawyers. They do not work in all areas of law, but out of responsibility for their clients, when demands arise in areas that they are not specialized in, they involve another partner lawyer. In this case, to maintain communication with the client through the same channel, it would be necessary to be able to include this partner lawyer only in these specific conversations.

Perhaps a guest profile with permission only to read/reply to messages where he was added as a participant would be enough.

**Reply by sanchez777** [reply]
From my experience, here are the permissions that are most needed.
High priority:
- As mentioned in my response above, option to allow between agents viewing all conversations, viewing those from same team, or viewing their own only, whilst admins would view them all regardless of option chosen
- Not allowing agents to delete incoming messages or other agents' messages
- Not allowing agents to close others' conversations

Low priority:
- Users changing their name/exhibition name (removing such would only leave admins in charge of that)
- Not allowing agents to reopen or send messages to closed conversations
- Not allowing agents to edit contact info and merge contacts

**Reply by neoand** [reply]
Dear team,

Thank you for reaching out and for the opportunity to discuss our situation regarding the “Custom Roles” feature. We would like to share our perspective to help you better understand why this feature is so important to us.

We are a non-profit organization dedicated to providing spiritual and psychological support to people who reach out through various contact methods such as WhatsApp, email, and web platforms. We have more than 600 volunteers who contribute to our cause by offering these services to the community. However, as an non-profit organization, our financial resources are limited, making it difficult for us to access paid enterprise solutions.

We understand that your focus is on providing these features for larger businesses, but given the nature of our organization and the large number of volunteers we manage, we require more precise role and permission management to ensure an organized and efficient service. For us, the ability to manage our volunteers through custom roles is not just about compliance; it’s an operational necessity to adequately assign tasks and track communications with the people who seek our help.

While it’s understandable that certain features are geared towards companies with more resources, we kindly request that you reconsider your decision to keep this feature as part of a premium or enterprise plan. We firmly believe that making it available to all, regardless of organization size, would greatly benefit not only organizations like ours but also any group that, despite having a large number of collaborators, lacks the financial means to access enterprise tools.

We hope you can understand our situation and appreciate any effort you make to support our mission and others like us.

Sincerely,
Anderson Gomes

**Reply by Korayem** [reply]
At least you could've said that this is a planned enterprise feature early on so can decide whether to utilize chatwoot or look else where instead of waiting and wasting time

**Comment by LuciaBeron3456** [comment]
Wow, it's paid? That sucks, it's kind of a dealbreaker for me, I was loving the tool

**Comment by fernandescontabilidade** [comment]
When will it be possible to send WhatsApp templates with the following parameters: head, body and footer?

Example:

{
  "paraments": [
    {
      "type": "head",
      "sub-type": "document",
      "1": "https://example.com/doc.pdf"
    },
    {
      "type": "body",
      "sub-type": "text",
      "1": "Hello, this is a message template for WhatsApp."
    },
    {
      "type": "footer",
      "sub-type": "button",
      "1": "Continue",
      "payload": "next_action_payload"
    }
  ]
}

**Comment by neoand** [comment]
Dear team,

Thank you for reaching out and for the opportunity to discuss our situation regarding the “Custom Roles” feature. We would like to share our perspective to help you better understand why this feature is so important to us.

We are a non-profit organization dedicated to providing spiritual and psychological support to people who reach out through various contact methods such as WhatsApp, email, and web platforms. We have more than 600 volunteers who contribute to our cause by offering these services to the community. However, as an non-profit organization, our financial resources are limited, making it difficult for us to access paid enterprise solutions.

We understand that your focus is on providing these features for larger businesses, but given the nature of our organization and the large number of volunteers we manage, we require more precise role and permission management to ensure an organized and efficient service. For us, the ability to manage our volunteers through custom roles is not just about compliance; it’s an operational necessity to adequately assign tasks and track communications with the people who seek our help.

While it’s understandable that certain features are geared towards companies with more resources, we kindly request that you reconsider your decision to keep this feature as part of a premium or enterprise plan. We firmly believe that making it available to all, regardless of organization size, would greatly benefit not only organizations like ours but also any group that, despite having a large number of collaborators, lacks the financial means to access enterprise tools.

We hope you can understand our situation and appreciate any effort you make to support our mission and others like us.

Sincerely,
Anderson Gomes

**Reply by sojan-official** [reply]
Hey Anderson,

Thank you so much for sharing the details with us. While we’re working on improving the custom roles feature based on your feedback, I wanted to let you know that we offer free and discounted plans for non-profits and open-source projects. If you’re interested, feel free to reach out to our support team at any time.

We’d be happy to help!

**Comment by eloimarquessilva** [comment]
I think this issue would be resolved very simply: Allowing premium resources for self-hosted versions to be hired per resource and not per agent, since in these cases the cost of more agents does not impact the cost of Chatwoot's hosting structure.

For example: 

$5/month for Custom Branding
$5/month for Agent Capacity
$5/month Roles & Permissions

This would generate an increase in recurring revenue to maintain the project without an increase in cost for the Chatwoot company and would still provide a sustainable scenario for those using the community version.

**Reply by sojan-official** [reply]
Thanks for the suggestion! A per-feature pricing model for self-hosted users might seem straightforward, but we’ve seen cases where users rebrand and resell Chatwoot, which directly competes with our cloud offering. This makes it challenging to implement such a model effectively. However, we are considering options like specific add-ons (e.g., custom branding) that can address certain use cases without undermining the sustainability of our cloud business.

**Reply by flachdaniel** [reply]
Hi @sojan-official !
Is is possible to pay a fixed rate for custom branding for self hosted instances, instead of a /agent model? 

This is our only issue to go with chatwoot at the moment.

Also, do custom branding enables us to remove the "Changelog" and "Read Documentation" menu list items in the User menu, without modifying and rebuilding the project ourselves ?

**Comment by ongtungduong** [comment]
If enabling Custom Roles isn’t feasible at the moment, at least ensure agents cannot view all conversations in the inbox. Basic features like this would make Chatwoot more appealing to a broader audience, creating opportunities to sell premium features later.

Locking such a fundamental feature behind a paywall might drive users to other more mature platforms. Improving these basics will help Chatwoot grow its user base and position itself better in the market.

**Comment by Mindula-Dilthushan** [comment]
> ## Changelog
> * Custom Roles (alpha) [EE]
> * Admin page redesigns for macros, automation, inbox management, etc
> * Support for copy-pasting images in help center articles
> * Update GPT model for open AI integration to 4o-mini
> * Numerous bug fixes and enhancements

Custom Roles (Alpha) [EE]
This might finally allow more granular conversation access controls in the Enterprise Edition (EE). If they expand this feature, restricting conversations per agent might become possible without custom modifications.

**Comment by Evoltstore** [comment]
Hi Chatwoot team

From the documentation on  [Custom Roles and Role-Based Access Control](https://chatwoot.help/hc/user-guide/articles/1741923706-manage-team-access-control-with-flexible-role_based-permissions) we understand that when assigning a custom role to an agent, the conversation visibility permissions (e.g. “Manage all conversations” or “Manage only assigned/unassigned”) are **applied globally across all inboxes** where that agent is a member.

This means there's no apparent way to configure **inbox-level visibility scopes**. For example:

> We'd like to allow a user to:
> - Have full visibility of all conversations in the Website Live Chat inbox
> - But only see assigned conversations in the Email Support inbox

From what we can tell, this isn’t currently possible as the permission scope for conversations is either fully open or restricted, **across all inboxes**.

This creates a real limitation when teams are structured by department or channel sensitivity:
- You might want shared inboxes (like live chat) to be entirely visible to assigned agents
- While keeping conversations in sensitive inboxes (e.g. internal HR support, sales, VIP clients) locked down unless explicitly assigned to them


 "Custom Roles" has this issue, permission scoping should be applied on Inbox/Channel level, and not on user level, 
 so a user could be for example an full access "Admin" in one channel/inbox, and "support agent" in another inbox with limited acces, and so on and so forth.

Is this something on your roadmap, or something you'd consider improving in future releases?

Thanks again for the great documentation and product 🙌

**Comment by rafaelpontoandrade** [comment]
Well... it's really frustrating to charge for basic privacy features. Disappointed and looking for alternatives.
